### PR TITLE
Restore old telemetry.

### PR DESCRIPTION
**Pull request by mcarare**
Notes:

1. This restores the old telemetry in the same state that it was when the migration to Glean process began. Meaning, if the previous implementation was not proper, this just restores it. Some of the obvious errors were fixed while restoring.
2. The tips telemetry has not been restored, the previous one even had a wrong trigger.
3. Some of the context menu/browser telemetry was not recorded after the migration to AC. I reintroduced the ones that are still valid.
4. Some telemetry was obsolete even before the migration, for example, switch to gecko view from the web view. This obsolete telemetry was not reinstated. 

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

**Review by ionutbedregeanu — APPROVED** [review]

File: app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt [hunk 1]
```diff
@@ -506,6 +507,9 @@ class BrowserFragment :
 
             DownloadState.Status.COMPLETED -> {
                 Downloads.downloadCompleted.record(NoExtras())
+
+                TelemetryWrapper.downloadDialogDownloadEvent(sentToDownload = true)
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt:511]: self note: we always send true only here?

File: app/src/main/java/org/mozilla/focus/fragment/CrashReporterFragment.kt [hunk 2]
```diff
@@ -24,10 +25,14 @@ class CrashReporterFragment : Fragment(R.layout.fragment_crash_reporter) {
 
         CrashReporter.displayed.record(NoExtras())
 
+        TelemetryWrapper.crashReporterOpened()
+
         binding.closeTabButton.setOnClickListener {
             val wantsSubmitCrashReport = binding.sendCrashCheckbox.isChecked
             CrashReporter.closeReport.record(CrashReporter.CloseReportExtra(wantsSubmitCrashReport))
 
+            TelemetryWrapper.closeTabButtonTapped(wantsSubmitCrashReport)
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/fragment/CrashReporterFragment.kt:34]: self note: correct event for _crash reporter_ close?

File: app/src/main/java/org/mozilla/focus/fragment/FirstrunFragment.kt [hunk 3]
```diff
@@ -62,6 +64,8 @@ class FirstrunFragment : Fragment(), View.OnClickListener {
             R.id.skip -> {
                 finishFirstrun()
                 Onboarding.skipButtonTapped.record(Onboarding.SkipButtonTappedExtra(currentItem))
+                TelemetryWrapper.skipFirstRunEvent()
+                TelemetryWrapper.finishFirstRunEvent()
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/fragment/FirstrunFragment.kt:68]: Shouldn't finish be below for `R.id.finish`?

**Review by jonalmeida — APPROVED** [review]
I left two self-notes for myself to check some events out that I can't recall and one question.

If this is blocking a release or experiment, feel free to land it and I'll get back to my notes later as a follow-up.

File: app/src/main/java/org/mozilla/focus/fragment/FirstrunFragment.kt (hunk 3 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/fragment/FirstrunFragment.kt:68]: Good catch! Thank you!

File: app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt (hunk 1 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt:511]: Yes, we send `false` when the download is canceled: https://github.com/mozilla-mobile/focus-android/blob/b34f1ebf56eb6dbb9f1ef214a994c7fca87fc72d/app/src/main/java/org/mozilla/focus/telemetry/TelemetryMiddleware.kt#L61-L66

**Pull request merged**